eth/db/journal.py

```diff
@@ -80,43 +81,86 @@ def record_changeset(self) -> uuid.UUID:
         self.journal_data[changeset_id] = {}
         return changeset_id
 
-    def pop_changeset(self, changeset_id: uuid.UUID) -> Dict[bytes, bytes]:
+    def pop_changeset(self, changeset_id: uuid.UUID) -> Dict[bytes, Union[bytes, DeletedEntry]]:
         """
         Returns all changes from the given changeset.  This includes all of
         the changes from any subsequent changeset, giving precidence to
         later changesets.
         """
         if changeset_id not in self.journal_data:
-            raise KeyError("Unknown changeset: {0}".format(changeset_id))
+            raise KeyError(changeset_id, "Unknown changeset in JournalDB")
 
         all_ids = tuple(self.journal_data.keys())
         changeset_idx = all_ids.index(changeset_id)
         changesets_to_pop = all_ids[changeset_idx:]
+        popped_clears = tuple(idx for idx in changesets_to_pop if idx in self._clears_at)
```

Comment:
It *appears* that we only want the last one based on line 98.  Can we invert this to iterate over `changesets_to_pop` in reverse order and stop at the first one we encounter that is truthy for `in self._clears_at`

Reply:
Unfortunately, we also use it here:
https://github.com/ethereum/py-evm/pull/1745/files#diff-4691bcc3798d57741a237f81721f0b81R118

```py
        self._clears_at.difference_update(popped_clears)
```

I can't think of better way to remove the old clears. Although it does remind me that we should empty the set on a `reset()`.